flycheck.el

```diff
@@ -5683,7 +5688,19 @@ and rely on Emacs' own buffering and chunking."
           (setq flycheck-temporaries nil)
           ;; Send the buffer to the process on standard input, if enabled.
           (when (flycheck-checker-get checker 'standard-input)
-            (flycheck-process-send-buffer process))
+            (condition-case err
+                (flycheck-process-send-buffer process)
+              ;; Some checkers exit before reading all input, causing errors
+              ;; such as a `file-error' for a closed pipe, or a plain “no longer
+              ;; connected to pipe; closed it” error for a disconnection.  We
+              ;; report them if needed in `flycheck-finish-checker-process' (see
+              ;; `https://github.com/flycheck/flycheck/issues/1278').
+              (error (process-put process 'flycheck-error err))))
+          ;; Set the actual sentinel and process any events that might have
+          ;; happened while we were sending input.
+          (setf (process-sentinel process) #'flycheck-handle-signal)
```

Comment:
Why do we need to set the sentinel back to `-handle-signal` in this case?

Reply:
A checker returning errors on the fly could return a few errors as we send it our input, and a few more after we're done, so we need to preserve the handler.